crates/sim2h/src/job/pool.rs

```diff
@@ -0,0 +1,55 @@
+use crate::*;
+
+/// a cpu-count pool of threads that can execute jobs
+pub(crate) struct Pool {
+    job_cont: Arc<Mutex<bool>>,
+    job_threads: Vec<std::thread::JoinHandle<()>>,
+    job_send: crossbeam_channel::Sender<Box<dyn Job>>,
+}
+
+impl Pool {
+    pub(crate) fn new() -> Self {
+        let (job_send, job_recv) = crossbeam_channel::unbounded::<Box<dyn Job>>();
+        let job_cont = Arc::new(Mutex::new(true));
+        let mut job_threads = Vec::new();
+        for _ in 0..num_cpus::get() {
+            let cont = job_cont.clone();
+            let send = job_send.clone();
+            let recv = job_recv.clone();
+            job_threads.push(std::thread::spawn(move || loop {
+                {
+                    if !*cont.f_lock() {
+                        return;
+                    }
+                }
+
+                if let Ok(mut job) = recv.try_recv() {
+                    if job.run() {
+                        send.send(job).expect("failed to send job");
```

Comment:
What happens if the thread dies? Are you tracking this somehow? The pool might need to recreate threads to the match the number of cores.

Reply:
Good point. For now, I'll make it so if a thread panics it will bring down the whole process - I almost feel like rust should behave that way by default anyways... We can look into more robust thread handling in the future.